internal/resources/otelcollector/agent.go

```diff
@@ -186,7 +187,11 @@ func (aad *AgentApplierDeleter) makeAgentDaemonSet(configChecksum string) *appsv
 	annotations := map[string]string{commonresources.AnnotationKeyChecksumConfig: configChecksum}
 	maps.Copy(annotations, makeIstioAnnotations(IstioCertPath))
 
-	podSpec := makePodSpec(aad.baseName, aad.image, aad.podOpts, aad.containerOpts)
+	// Add pod options shared between all agents
+	podOpts := slices.Clone(aad.podOpts)
+	podOpts = append(podOpts, commonresources.WithTolerations(commonresources.CriticalDaemonSetTolerations))
```

Comment:
Since we have a pointer to the `aad`, can't we directly append the tolerations to the `aad.podOpts` without the need of cloning it?

Reply:
I did it to fix this violation https://go-critic.com/overview.html#appendassign